### ISSUE TITLE
Engine initialization requires two calls

### ISSUE THREAD

**Issue by birchmd** [aurora-is-near/aurora-engine#161]
During its development, the eth-connector was essentially treated as a separate contract that happens to be contained inside the main engine. A side effect of this is that it has its [own initialization function](https://github.com/aurora-is-near/aurora-engine/blob/50bf61cf7036e20675010c7399428de08f2e92ba/src/lib.rs#L443), which is separate from [the one for the engine](https://github.com/aurora-is-near/aurora-engine/blob/50bf61cf7036e20675010c7399428de08f2e92ba/src/lib.rs#L110) itself.

However, the connector is not a distinct contract, it is just a part of the engine and having to make two calls to fully initialize the engine is sub-optimal, for example this recently caused an [issue with the bully](https://github.com/aurora-is-near/aurora-engine/issues/122#issuecomment-865438367).

The purpose of this issue is to make the call to `new` alone enough to initialize the contract.

Additionally, there are two prover-related fields: [one in the connector](https://github.com/aurora-is-near/aurora-engine/issues/122#issuecomment-865438367) and [one in the engine state](https://github.com/aurora-is-near/aurora-engine/blob/50bf61cf7036e20675010c7399428de08f2e92ba/src/engine.rs#L157). The latter appears to never be used so I suspect that it could be removed as it is likely a duplicate of the former. This is relevant to this issue because it means the initialization still only needs to take one prover argument.

Note: any change to the contract initialization also needs to be reflected in the `aurora.js` package. For example, see https://github.com/aurora-is-near/aurora.js/pull/3

**Comment by sept-en** [comment]
@birchmd  this is a good suggestion! The thing worth noticing here is that `new()` for the engine allows setting the state for it multiple times, however, eth-connector allows doing the initialization only once. This is because obviously, we don't want to accidentally overwrite the `FungibleToken` data. For the sake of updating the contract addresses in eth-connector, I added some time ago `set_eth_connector_contract_data()` method which sets the provided prover ID and `EthCustodian` address.
So I think the correct behavior for eth-connector in `new()` will be the following:
* If eth-connector is not initialized -> call `init_contract()` method;
* If eth-connector is initialized -> call `set_contract_data()` method.

For sure, we can modify the behavior of `init_contract` allowing it to be called multiple times without modifying `FungibleToken` and `paused_mask` if the contract is initialized, but I think this will be a little bit misleading, so I'd prefer having the check on the engine level in `new()` method.

For the prover ID in the engine state, I remember I reported about it some time ago and I even remember that I removed it. Not sure why we still have it there, but this is quite minor so we can just proceed with removing it again.

**Comment by mrLSD** [comment]
@birchmd  What practical case for **mainnet** use multiple times `new` function?
Also as mentioned @sept-en `FunginlrToken` should be inited only once.

I think we can add additional parameters for `new` function for `eth-connector` part. And again check for only one eth-connector init call.

**Comment by birchmd** [comment]
> What practical case for mainnet use multiple times new function?

I don't think this is an important consideration. We want to be able to deploy the contract more places than just mainnet. For example, community members may wish to deploy it on a local node for testing purposes. And getting an error because they only called `new` without `new_eth_connector` is a bad developer experience. If they use the aurora CLI then this complexity is hidden because there is a single `init` call there. But I think the point remains that it is unintuitive to have two required calls when it would be easy enough for us to only have one.
